fili-core/src/test/groovy/com/yahoo/bard/webservice/table/PhysicalTableSpec.groovy

```diff
@@ -150,4 +150,14 @@ class PhysicalTableSpec extends Specification {
         expect:
         physicalTable.getDimensions() == [dimension] as Set
     }
+
+    def "test physical to logical mapping is constructed correctly"() {
+        setup:
+        def oneDimPhysicalTable = new PhysicalTable("test table", DAY.buildZonedTimeGrain(UTC), ['dimension':'druidDim'])
+        def twoDimPhysicalTable = new PhysicalTable("test table", DAY.buildZonedTimeGrain(UTC), ['dimension1':'druidDim', 'dimension2':'druidDim'])
```

Comment:
At least in the context of our project, it's considered good practice (I don't know enough about the larger Groovy community to know if that's true more generally). 

If nothing else, it improves IDE support, because the IDE knows what the variable type is. 

It can also make things more readable, because it's easy for readers to see what the variable type this (though this can be partially offset because `def` is more concise, so which is more readable is going to be a matter of opinion). 

Duck typing has its advantages, but it has its disadvantages too. Here we're not using any of the advantages, but we're still saddled with the disadvantages.